playbooks/build-oc-images.yml

```diff
@@ -0,0 +1,60 @@
+# This playbook is not included in "ir-XYZ" flows as is not
+# related to any of them and is used only stand-alone.
+# 
+# This playbook is designed to run on single node in virtual
+# machine environment. Currently node called "controller"
+# is used as main node doing this task as IR doesn't support
+# creating stand-alone single node topology and we still
+# need to have this playbook integrated with current IR workflow.
+---
+- name: Store local public key as variable for current user
+  hosts: controller
+  gather_facts: no
+  tasks:
+    - name: Create and save public key from private key
+      shell: "ssh-keygen -y -f {{ hostvars[inventory_hostname].ansible_ssh_private_key_file }}"
+      register: cmd_priv_key
+      delegate_to: localhost
+
+- name: Enable root access as cloud images might not have one first
+  hosts: controller
+  gather_facts: no
+  become: yes
+  become_user: root
+  tasks:
+    - name: Add local public key to root's authorized_keys
+      authorized_key:
+        key: "{{ cmd_priv_key.stdout }}"
+        user: root
+
+    - name: Copy local private key to remote root
+      copy:
+          src: "{{ hostvars[inventory_hostname].ansible_ssh_private_key_file }}"
+          dest: "/root/.ssh/id_rsa"
+
+    - name: Copy local public key to remote root
+      copy:
+          content: "{{ cmd_priv_key.stdout }}"
+          dest: "/root/.ssh/id_rsa.pub"
+
+    - name: Fix permissions for both keys
+      shell: |
+            chmod 750 /root/.ssh/id_rsa
+            chmod 700 /root/.ssh/id_rsa
+
+- name: Build images as root and upload them to remote storage
+  hosts: controller
+  gather_facts: yes
+  become: yes
+  become_user: root
+  roles:
+    - role: installer/ospd/images
+      task: 'build'
```

Comment:
Why should be default values inside role pre-filled by private.somekey values? It's definitely not a good idea.
If I want to use something, that is designed to be reusable, it's my own business to pass proper values as I need to get desired behavior. It doesn't matter if it's me as 'user', or 'author' of this file. The author is kind of 'user' as well.
Role is like a 'function' in any ordinary language with ability to define arguments. If there would be set some default values, it should be chosen very carefully and not force user to redefine them to 'empty values to get basic functionality.

In short, I definitely agree with @fhubik and vote +1 for implementation in this patch.